DirectProgramming/DPC++FPGA/Tutorials/Features/experimental/register_map_and_streaming_interfaces/src/register_map_and_streaming_interfaces.cpp

```diff
@@ -0,0 +1,209 @@
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/ext/intel/prototype/interfaces.hpp>
+#include <sycl/sycl.hpp>
+
+#include "exception_handler.hpp"
+
+using ValueT = int;
+// Forward declare the kernel names in the global scope.
+// This FPGA best practice reduces name mangling in the optimization reports.
+class LambdaRegisterMapControlIP;
+class LambdaStreamingControlIP;
+
+// offloaded computation
+ValueT SomethingComplicated(ValueT val) { return (ValueT)(val * (val + 1)); }
+
+/////////////////////////////////////////
+
+struct FunctorStreamingControlIP {
+  // Use the 'conduit' annotation on a kernel argument to specify it to be
+  // a streaming kernel argument.
+  conduit ValueT *input;
+  conduit ValueT *output;
+  // Without the annotations, kernel arguments will be inferred to be streaming
+  // kernel arguments if the kernel control interface is streaming, and
+  // vise-versa.
+  size_t n;
+  FunctorStreamingControlIP(ValueT *in_, ValueT *out_, size_t N_)
```

Comment:
this constructor is not needed since this `struct` is treated as an [aggregate](https://en.cppreference.com/w/cpp/language/aggregate_initialization).